### PR TITLE
refactor: Hoist PoolClient off UMA dependencies

### PR DESCRIPTION
**Pull request by pxrl**
This required importing the following components from the UMA repo:
 - RateModel + RateModelDictionary
 - TransactionManager

Some post-import modifications were needed in the TransactionManager
in order to adjust it from a closure to a class.

### CODE REVIEW

**Comment by pxrl** [comment]
Note: This PR has not been fully tested yet.

**Comment by james-a-morris** [comment]
Is this PR ready for review? Happy to do so if it's good to go 🙂

**Comment by pxrl** [comment]
> Is this PR ready for review? Happy to do so if it's good to go 🙂

Thanks for the nudge ! It builds OK now that its dependencies have all been merged. I still haven't executed any manual test to verify the output, but I think the code itself is fine for review.

**Review by nicholaspai — APPROVED** [review]

File: src/pool/poolClient.ts [hunk 1]
```diff
@@ -628,7 +645,7 @@ export class Client {
     const txman = this.getOrCreateTransactionManager(signer, userAddress);
 
     const request = await contract.populateTransaction.addLiquidity(l1Token, l1TokenAmount, overrides);
-    const id = await txman.request(request);
+    const id = txman.request(request);
```
Comment (james-a-morris) [inline src/pool/poolClient.ts:648]: Are these no longer async?

File: src/pool/poolClient.ts [hunk 2]
```diff
@@ -777,15 +794,15 @@ export class Client {
     await this.updateAndEmitUser(userState, poolState, poolEventState);
   }
   async updatePool(l1TokenAddress: string, overrideLatestBlock?: Block): Promise<void> {
-    // default to 100 block delta unless specified otherwise in config
-    const { blockDelta = DEFAULT_BLOCK_DELTA } = this.config;
+    // Lookup exchange rate based on 10 block delta unless overridden by config
+    const { blockDelta = 10 } = this.config;
```
Comment (james-a-morris) [inline src/pool/poolClient.ts:798]: Should we maintain this as a constant elsewhere or is this the only usage

**Review by james-a-morris — COMMENTED** [review]
Left some comments

File: src/pool/poolClient.ts [hunk 3]
```diff
@@ -111,6 +116,17 @@ export type PooledToken = {
   undistributedLpFees: BigNumber;
 };
 
+// Loop forever but wait until execution is finished before starting next timer. Throw an error to break this
+// or add another utlity function if you need it to end on condition.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function loop(fn: (...args: any[]) => any, delay: number, ...args: any[]) {
```
[the remark this answers is not in the document]
Comment (pxrl) [reply src/pool/poolClient.ts]: It looks like it's intended to run indefinitely in the background, so imposing some max loop count might break the expected behaviour. FWIW I'm not really sure how this whole thing works - there are _lots_ of lines of implementation here, so I'm a bit reluctant to modify it unless we find an issue.

https://github.com/across-protocol/sdk-v2/blob/eed49828bb48331d915240e628024da1f6944ffc/src/pool/poolClient.ts#L826

**Review by james-a-morris — APPROVED** [review]
LGTM

File: src/pool/poolClient.ts [hunk 4]
```diff
@@ -1,28 +1,33 @@
 import assert from "assert";
+import { ExpandedERC20__factory as erc20 } from "@across-protocol/contracts-v2";
```
[the remark this answers is not in the document]
Comment (pxrl) [reply src/pool/poolClient.ts:2]: @james-a-morris ooc, do you know if this is still likely to be an issue after your recent packaging work?

**Comment by pxrl** [comment]
Parking this for now because it's not used in the bots and there is probably breakage in the FE. Will pick it up later.

**Pull request closed without merging**